scripts/zones/Apollyon/mobs/Flying_Spear.lua

```diff
@@ -2,19 +2,79 @@
 -- Area: Apollyon SE
 --  Mob: Flying Spear
 -----------------------------------
-require("scripts/globals/limbus");
------------------------------------
-
-function onMobDeath(mob, player, isKiller)
-end;
+require("scripts/globals/pathfind")
+local ID = require("scripts/zones/Apollyon/IDs")
+local flags = tpz.path.flag.NONE
+local path =
+{
+    [1] = 
+    {
+        {509.874, 0.000, -310.010},
+        {490.000, 0.000, -287.000}
+    },
+    [2] = 
+    {
+        {559.858, 0.000, -347.135},
+        {540.000, 0.000, -347.000}
+    },
+    [3] = 
+    {
+        {597.587, 0.000, -348.038},
+        {619.000, 0.000, -337.000}
+    },
+    [4] = 
+    {
+        {600.347, 0.000, -358.427},
+        {599.808, 0.000, -377.934}
+    },
+    [5] = 
+    {
+        {525.602, 0.000, -353.234},
+        {541.000, 0.000, -368.000}
+    },
+    [6] = 
+    {
+        {539.524, 0.000, -352.514},
+        {559.506, 0.000, -349.762}
+    },
+    [7] = 
+    {
+        {613.174, 0.000, -357.948},
+        {615.098, 0.000, -377.595}
+    },
+    [8] = 
+    {
+        {570.718, 0.000, -343.563},
+        {564.009, 0.000, -314.711}
+    }
+}
 
-function onMobDespawn(mob)
-     local mobID = mob:getID();
-     if (mobID == 16933034) then -- time
+function onMobRoam(mob)
+    local offset = mob:getID() - ID.mob.APOLLYON_SE_MOB[4]
+    local pause = mob:getLocalVar("pause")
+    if pause < os.time() then
+        local point = (mob:getLocalVar("point") % 2)+1
+        mob:setLocalVar("point", point)
+        mob:pathTo(path[offset][point][1], path[offset][point][2], path[offset][point][3], flags)
+        mob:setLocalVar("pause", os.time()+15)
+    end
+end
 
-     elseif (mobID == 16933037) then -- recover
+function onMobSpawn(mob)
+    mob:setMod(tpz.mod.UDMGMAGIC, -100)
+end
 
-     elseif (mobID == 16933040) then -- item
-
-     end
-end;
+function onMobDeath(mob, player, isKiller)
+    if isKiller then
+        local battlefield = player:getBattlefield()
+        battlefield:setLocalVar("killCountF4", battlefield:getLocalVar("killCountF4")+1)
+        local killCount = battlefield:getLocalVar("killCountF4")
+        GetMobByID(ID.mob.APOLLYON_SE_MOB[4]):setMod(tpz.mod.UDMGPHYS, -(8-killCount)*10)
+        if killCount == 1 then
+            GetNPCByID(ID.mob.APOLLYON_SE_MOB[4]):setStatus(tpz.status.DISAPPEAR)      
+            GetMobByID(ID.mob.APOLLYON_SE_MOB[4]):spawn()
```

Comment:
Does the floor boss not spawn without a Flying Spear being killed?

Reply:
It's an NPC until someone kills one of the Flying Spears.

Reply:
Oh, you told me about this one and I forgot. Sorry!